hedera-mirror-web3/src/main/java/com/hedera/mirror/web3/repository/EntityRepository.java

```diff
@@ -38,6 +39,21 @@ public interface EntityRepository extends CrudRepository<Entity, Long> {
             unless = "#result == null")
     Optional<Entity> findByEvmAddressAndDeletedIsFalse(byte[] alias);
 
+    @Cacheable(
+            cacheNames = CACHE_NAME_ALIAS,
+            cacheManager = CACHE_MANAGER_ENTITY,
+            key = "T(java.util.Arrays).hashCode(#alias)",
+            unless = "#result == null")
+    @Query(
+            value =
+                    """
+            select *
+            from entity
+            where (evm_address = ?1 or alias = ?1) and deleted is not true
+            """,
+            nativeQuery = true)
+    Optional<Entity> findByEvmAddressOrAlias(byte[] alias);
```

Comment:
At least right now we only need the entity.id field returned so we can optimize this and the other query.
```suggestion
    Optional<Long> findByEvmAddressOrAlias(byte[] alias);
```

Reply:
That's correct, we can simplify the query.

Reply:
This method is now used in `CommonEntityAccessor#get` as well and we need the full entity in there. We can have 2 separate queries with different return types. However, I am not sure if we will gain any performance this way.